src/Paysera/WalletApi/Mapper.php

```diff
@@ -2387,4 +2403,51 @@ public function encodeTransferConfiguration(Paysera_WalletApi_Entity_TransferCon
     {
         return (new Paysera_WalletApi_Mapper_TransferConfigurationMapper())->mapFromEntity($transferConfiguration);
     }
+
+    /**
+     * Decodes client wallet permissions
+     *
+     * @param array $data
+     *
+     * @return Paysera_WalletApi_Entity_ClientWalletPermissions
+     */
+    public function decodeClientWalletPermissions(array $data)
+    {
+        $scopes = [];
+
+        $walletPermissions = Paysera_WalletApi_Entity_ClientWalletPermissions::create()
+            ->setWalletId($data['id'])
+            ->setAccountNumber($data['account'])
+        ;
+
+        foreach ($data as $key => $value) {
+            switch ($key) {
+                case Paysera_WalletApi_Entity_ClientWalletPermissions::SCOPE_BALANCE:
+                case Paysera_WalletApi_Entity_ClientWalletPermissions::SCOPE_STATEMENTS:
+                    if ($value) {
+                        $scopes[] = $key;
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        $walletPermissions->setScopes($scopes);
+
+        return $walletPermissions;
+    }
+
+    public function encodeClientWalletPermissions(Paysera_WalletApi_Entity_ClientWalletPermissions $walletPermissions)
+    {
+        if (!$walletPermissions->getWalletId()) {
```

Comment:
$walletPermissions->getWalletId() === null, if walletId will be 0, it's still id, that's the reason why we should always use strict comparisons